tools/configure.py

```diff
@@ -20,21 +22,63 @@ def register(command):
         default=False,
         help="perform an interactive configuration using ccmake",
     )
+
+    command.add_argument(
+        "-r",
+        "--enable-roles",
+        action="store",
+        dest="enabled_roles",
+        default=None,
+        help="Colon-separated list of roles to enable. Pass 'all' to enable all.",
+    )
+
+    command.add_argument(
+        "-R",
+        "--disable-roles",
+        action="store",
+        dest="disabled_roles",
+        default=None,
+        help="Colon-separated list of roles to disable. If '-R' is set, '-r' is ignored",
+    )
+
     command.add_argument("args", nargs="...", help="the arguments to pass through to cmake")
 
 
 @run_on_docker
-def run(interactive, args, **kwargs):
+def run(interactive, enabled_roles, disabled_roles, args, **kwargs):
     pty = WrapPty()
 
-    # If interactive then run ccmake else just run cmake
+    # Change into the build directory
     os.chdir(os.path.join(b.project_dir, "..", "build"))
 
+    # If specific roles are disabled, then we enable all of the others
+    if disabled_roles:
+        disabled_roles = disabled_roles.strip()
+        # roles are a colon separated list, so we split on that character
+        disabled_roles = disabled_roles.split(":")
+        cprint("Disabling all roles except these: ", end=" ", color="red", attrs=["bold"])
+        cprint(str(disabled_roles), color="green", attrs=["bold"])
```

Comment:
`from termcolor import colored`
```suggestion
        desc_str = colored("Disabling all roles except these:", color="red", attrs=["bold"])
        roles_str = colored(", ".join([role for role in disabled_roles]), color="green", attrs=["bold"])
        print(f"{desc_str} {roles_str}")
```